server/controller/src/main/java/ai/starwhale/mlops/domain/swds/SWDatasetService.java

```diff
@@ -286,4 +287,16 @@ private List<SWDatasetInfoVO> swDatasetInfoOfDs(SWDatasetEntity ds) {
         return swDatasetVersionEntities.parallelStream()
             .map(entity -> toSWDatasetInfoVO(ds, entity)).collect(Collectors.toList());
     }
+
+    public String query(UploadRequest uploadRequest) {
+        SWDatasetEntity entity = swdsMapper.findByNameForUpdate(uploadRequest.name());
```

Comment:
we do not lock the row when in pure query process, so I think `findByName` is sufficent